### DISCUSSION TITLE
Fixed sidebar

### DISCUSSION THREAD

**Discussion by LITUATUI** [chipzoller/hugo-clarity#276] [Q&A] [answered]
Hi.

What's the best way to make the sidebar fixed?

Thanks.

**Comment by chipzoller** [comment]
Please explain.

**Reply by LITUATUI** [reply]
I tried with custom CSS to make the sidebar fixed or sticky, but with poor results. Is there a field to enable it in the config.toml file? From my experience with WordPress a fixed or sticky sidebar is great place to put a 300x250 banner.

I'm trying to migrate a [blog](https://pushevs.com) from WordPress and make it look similar in Hugo. Your theme and [HBS](https://hbs.razonyang.com/en/) are the best themes I found so far for what I want.

**Comment by onweru** [comment, marked as answer]
@LITUATUI, in my experience, if the sidebar is taller than the screen height, making it sticky works but not smoothly. There's no knob on the config to enable that. It's not that hard to implement though.

I brought up the issue of the sidebar height because in most cases, the sidebar will be taller than the client's height. In which case it would be much better to make part of the sidebar sticky.

For demo purposes, I will a recording of a site that uses this theme with its sidebar being sticky.

https://user-images.githubusercontent.com/16350351/160151211-56550657-3c4c-4d9d-88fc-5458cb7997b0.mov

If you insist on having the entire sidebar being sticky, these are the styles you should add to your custom CSS file

```sass
// assets/sass/_custom.sass
.sidebar
  &_inner
    position: sticky
    position: -webkit-sticky
    top: 60px
```

**Reply by LITUATUI** [reply]
Thanks @onweru , that works great.

This theme now only needs a search function that works (maybe I'm doing it wrong) and a prev/next post navigation at the end of every post.

**Reply by onweru** [reply]
You're most welcome 🤝. Please check the readme, I added a search function yesterday.

**Reply by LITUATUI** [reply]
I did, but always get "No results found".

**Reply by onweru** [reply]
@LITUATUI, is your site available publicly? Some sort of a preview since you are in the middle of a migration

**Reply by LITUATUI** [reply]
@onweru sorry, not yet. I still have a lot of work to do before it's ready to upload to GitHub. 

Right now I have been trying it locally with this theme and [HBS](https://hbs.razonyang.com/en/). Both seem very promising with only one thing or two missing. 

Search not working with Clarity:

![pushevs-clarity](https://user-images.githubusercontent.com/49299328/160294070-17b0d267-b6ae-4ee9-802d-ed1c60483087.jpg)

Search working with HBS:

![pushevs-hbs](https://user-images.githubusercontent.com/49299328/160294153-8c737bd0-4292-4b8a-9e0d-8a6780644abe.jpg)

The RSS feed is working and index.xml is available, this is what is used to search keywords right?

**Reply by onweru** [reply]
index.json yeah. Most likely `mainSections` under params.toml file isn't set if you followed the other instructions in the READme

**Reply by LITUATUI** [reply]
Thanks. That was it :)

I had this:

`mainSections = ["post"]`

Changed to this:

`mainSections = ["posts"]"`

Search is working now and faster than with HBS theme. Sorry for being a noob.
